sources/describe.py

```diff
@@ -0,0 +1,138 @@
+import logging
+
+from Parser.ArgParser import ArgParser
+from Parser.CSVParser import CSVParser
+from math import sqrt, floor, ceil
+
+
+logging.getLogger().setLevel(logging.INFO)
+
+
+class Describe(CSVParser):
+    """
+    Take a dataset in parameter and can display some information about it.
+    """
+    content_vars: dict
+
+    def __init__(self, args: ArgParser):
+        super().__init__(args)
+        self.desc_vars = {
+            "count": len,
+            "mean": self._mean,
+            "std": self._std_dev,
+            "min": self._min,
+            "0.25": self._percentile,
+            "0.50": self._percentile,
+            "0.75": self._percentile,
+            "max": self._max,
+            "unique": self._unique,
+        }
+        self.csv_parser()
+        self._set_val()
+
+    @staticmethod
+    def _percentile(array, percent=0.50):
+        # can reduce sorting time by getting already sorted array
+        array = array.sort_values(kind="mergesort").values
+        try:
+            percent = float(percent)
+            k = (len(array) - 1) * percent
+            f = floor(k)
+            c = ceil(k)
+            if f == c:
+                return array[int(k)]
+            d0 = array[int(f)] * (c - k)
+            d1 = array[int(c)] * (k - f)
+            return d0 + d1
+        except (ValueError, TypeError):
+            return "NaN"
+
+    @staticmethod
+    def _unique(array):
+        unique = {}
+        for value in array:
+            if value not in unique:
+                unique[value] = True
+        return len(unique)
+
+    @staticmethod
+    def _max(array):
+        maximum = array[0]
+        for value in array:
+            if value > maximum:
+                maximum = value
+        return maximum
+
+    @staticmethod
+    def _min(array):
+        minimum = array[0]
+        for value in array:
+            if value < minimum:
+                minimum = value
+        return minimum
+
+    @staticmethod
+    def _variance(data):
+        n = len(data)
+        try:
+            mean = sum(data) / n
```

Comment:
call self._mean() here

Reply:
no cause it will recalcul the len of the data twice, so i'd prefer doing it like this, and also it's better for the try catch to don't call self._mean

Reply:
i create a optionnal argument in _mean for the len of the array

Reply:
finally i don't, i prefer keep this function static (_variance).

Reply:
ok for the static to keep a continuity in the class but a len calculus isn t that big